tests/acceptance/features/bootstrap/FeatureContext.php

```diff
@@ -1785,6 +1796,10 @@ public function getEmailAddressForUser($userName) {
 	 * @return string
 	 */
 	public function getActualUsername($functionalUsername) {
+		$usernames = $this->usersToBeReplaced();
+		if (isset($usernames, $usernames[$functionalUsername])) {
+			return $usernames[$functionalUsername];
+		}
```

Comment:
```suggestion
		if (isset($usernames)) {
			if (isset($usernames[$functionalUsername])) {
				return $usernames[$functionalUsername];
			}
			$normalizedUsername = $this->normalizeUsername($functionalUsername);
			if (isset($usernames[$normalizedUsername])) {
				return $usernames[$normalizedUsername];
			}
		}
```
Some test scenarios check that different mixes of uppercase and lowercase in the username work - e.g. `user0` `User0` and `USER0` should be able to mixed up in API calls etc.

If somebody specifies an override for each of these, then we can replace each of them individually (e.g. `user0` =" `somebody` `User0` => `SomeBody` `USER0` => `SOMEBODY`). But if they do not want to do that, then we can automatically check and use the replacement for the normalized value of the username.

Reply:
fixed